crates/lsp/src/lsp.rs

```diff
@@ -164,6 +164,33 @@ struct Error {
     message: String,
 }
 
+#[derive(Debug)]
+pub enum ServerStatus {}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub enum ServerHealthStatus {
+    Ok,
+    Warning,
+    Error,
+    Other(String), // Variant to handle unknown values due to this still being experimental
```

Comment:
Can we promote this to a doc comment on the `Other` variant instead of having it be a line comment like this?